api/client/build.go

```diff
@@ -131,10 +130,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		return fmt.Errorf("cannot canonicalize dockerfile path %s: %v", relDockerfile, err)
 	}
 
-	var includes = []string{"."}
+	var (
+		includes = []string{"."}
+		excludes []string
+	)
 
-	excludes, err := utils.ReadDockerIgnore(path.Join(contextDir, ".dockerignore"))
-	if err != nil {
+	f, err := os.Open(filepath.Join(contextDir, ".dockerignore"))
+	switch {
```

Comment:
thanks will fix, can't remember why i used a switch here, maybe the code changed.